core/controllers/profile_test.py

```diff
@@ -236,9 +238,16 @@ def test_get_profile_picture_valid_username(self):
         response_dict = self.get_json(
             '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
         )
-        # Although the user has a valid username, they have not yet supplied
-        # a profile picture.
-        self.assertIsNone(
+        self.assertIsNotNone(
+            response_dict['profile_picture_data_url_for_username'])
+
+    def test_get_signup_gravatar(self):
+        self.signup(self.EMAIL, self.USERNAME)
+        response_dict = self.get_json(
+            '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
+        )
+        self.assertEqual(
+            self.get_gravatar_from_url(self.GRAVATAR_URL),
```

Comment:
@seanlip How about checking if the profile_picture_image_url starts with the string `data :' image/png'` so that we can be sure there's an image.
Or
We can mimic the exact functionality of urlfetch using urllib and then compare the results